### DISCUSSION TITLE
Updating from Phalcon 1.3.4

### DISCUSSION THREAD

**Discussion by Jorge22f** [phalcon/cphalcon#15254] [General]
I have a project that is using version 1.3.4. What are things I need to consider before updating? This is also running PHP 5.6.40 so I'm assuming I'd have to update that as well. Has Phalcon changed over time to the point where I'll need to make a lot of changes to the project code? if so at which version did this happen?

**Comment by talal424** [comment]
if you are planning on using php 8 you may want to wait for phalcon 5
either way you have to go through the [changes log](https://github.com/phalcon/cphalcon/blob/master/CHANGELOG.md)

**Comment by Jeckerson** [comment]
Hello, you are using very old version. Things changed pretty significant from your version. Most of are minor changes, which need to be checked case by case.

You could try to upgrade first to version 3, and use this automatic tool to check if all things are compatible with version 4. As phalcon 3 is supported on PHP5.
https://github.com/ruudboon/phalcon-docker-rector

After that, upgrade to phalcon `4.1` and PHP 7.4.

Support for PHP 8 will be done available in
* Phalcon 4.2 as extension
* PHP 5.0 as composer package

**Reply by AliShareei** [reply]
@jackerson, you mean Phalcon 5.0 as composer package, right?
As I found out from my previous QA, versions 4.x and 5.0 will be exactly the same. 
But if the server has a Phalcon 4.x extension at the same time and also the project included version 5 with composer, will not be caused any error?

**Reply by jackerson** [reply]
You meant to @jeckerson, I am @jackerson :)

**Reply by Jeckerson** [reply]
@AliShareei yes. If there will be both: ext and composer package, then you will need to choose as it will generate conflict in classes.
